### ISSUE TITLE
Question for  training details

### ISSUE THREAD

**Issue by terrychenism** [shicai/MobileNet-Caffe#1]
Could you share the training details? something like lr_step, wd and data aug?

**Comment by shicai** [comment]
we use inception data aug, base_lr=0.1, and wd=1e-4, which is similar to train resnet.
you can refer to: https://github.com/facebook/fb.resnet.torch

**Comment by terrychenism** [comment]
Thanks!  decrease lr by every 30 epoch?

**Comment by shicai** [comment]
we use linear policy, which performs slightly better than step=30 policy from my own experience.

**Comment by terrychenism** [comment]
Great! Thanks a lot!

**Comment by boluoweifenda** [comment]
@shicai 
Sorry to reopen this issue. May I ask more training details? I have tuned hyper-parameters a lot but only got about 30.5% error rate, seems still a long way: 
1. what optimizer do you use?  the original paper use RMSProp: what about the decay, momentum, and epsilon?
2. batch normalization details: momentum and decay, In tensorflow slim implementation the momentum=0.9997, seems to high for test accuracy
3. batchsize, I use 256 right now
4. weight decay: do you add weight decay of 1e-4 to all the parameters including depth_wise conv? In tensorflow slim they use 4e-5 and have no decay on depth_wise conv.
5. lr: 0.1->0.001 linearly after 60 epochs?

**Comment by XiongweiWu** [comment]
@boluoweifenda As a matter of fact, I found the provided model of author also produce this result(224/256, acc: 69.5/89.2). 
  
 BTW,  how about your training time? I use depthwise conv provided by others since I guess use caffe engine is very slow for training

**Comment by boluoweifenda** [comment]
@XiongweiWu I use tensorflow, 2 titanxp , about 2iter/s (batchsize 256), the utilization rate of GPU is about 100%

**Comment by mathmanu** [comment]
I am looking for references to MobileNetV2 training in few (eg. 90 epochs as usual). The following are few references that I could find (other than tensorflow), that provides atleast a part of training script with accuracy obtained. Let me know if any one finds a full training example.

* https://github.com/Randl/MobileNetV2-pytorch
They use total 400 epochs, lr=0.01 with step by 0.1 after 200 epochs,  batch_size = 64.
That takes too long time to train. I am looking for a faster training configuration.

* https://github.com/apache/incubator-mxnet/blob/master/docs/api/python/gluon/model_zoo.md
https://raw.githubusercontent.com/dmlc/web-data/master/gluoncv/logs/classification/imagenet/mobilenetv2_1.0.sh
I could find the model and script in the following mxnet model-zoo repo as well. It uses 480 epochs to train and their hyper parameters are similar to what is described in the mobilenetv2 paper (except for the use of default of optimizer, which I assume to be SGD).

* https://github.com/RuiminChen/Caffe-MobileNetV2-ReLU6/issues/1
Found a reference to training with roughly 90 epochs here:
But not the full training script.

**Comment by mathmanu** [comment]
I could find the model script in the following mxnet model-zoo repo as well. It uses 480 epochs to train and their hyper parameters are similar to what is described in the mobilenetv2 paper (except for the use of default of optimizer, which I assume to be SGD).

https://github.com/apache/incubator-mxnet/blob/master/docs/api/python/gluon/model_zoo.md

https://raw.githubusercontent.com/dmlc/web-data/master/gluoncv/logs/classification/imagenet/mobilenetv2_1.0.sh

**Comment by ujsyehao** [comment]
Can anyone train a high accuracy model?
I am trying to change some hyperparameters.

**Comment by andrew-wja** [comment]
Can somebody post a solver.prototxt that actually works, for others to use as a baseline? 

I have tried using every configuration mentioned in the issue comments in this repo, and I still cannot get the models to converge. If someone has a working solver configuration, could you please copy the entire solver config file so others can use it as a base? Thank you!
